database/database.go

```diff
@@ -77,3 +77,46 @@ func InsertUserProfile(user *model.UserProfile) error {
 	}
 	return err
 }
+
+//GenerateToken func to add token in db
+func GenerateToken(username string) error {
+	var userAuth *model.UserAuth
+	var hashedtoken string
+	var err error
+	hashedtoken, err = EncryptPassword(username)
+	if err == nil {
+		userAuth.Token = hashedtoken
+	} else {
+		log.Print(err)
+		return err
+	}
+	userAuth.Username = username
+	stmt, err := db.Prepare("INSERT INTO user_auth(Username,Token) VALUES(?,?)")
+	if err != nil {
+		log.Print(err)
+		return err
+	}
+	defer stmt.Close()
+	_, err = stmt.Exec(userAuth.Username, userAuth.Token)
+	if err != nil {
+		log.Print(err)
+		return err
+	}
+	return nil
+}
+
+//LoginUser func for sign_in
+func LoginUser(user *model.UserProfile) error {
+	var profile *model.UserProfile
+	var err error
+	profile, err = GetUserByUsername(user.Username)
+	if err == nil {

```

Comment:
Return err is if err ~= nil or profile is nil.
continue with rest code.

Login user should also return generatedToken